common/spec/dependabot/notices_spec.rb

```diff
@@ -23,6 +24,23 @@ def initialize(name:, version:, deprecated_versions: [], unsupported_versions: [
   attr_reader :unsupported_versions
   attr_reader :supported_versions
   attr_reader :support_later_versions
+
+  sig { override.returns(T::Boolean) }
+  def deprecated?
+    # If the version is unsupported, the unsupported error is getting raised separately.
+    return false if unsupported?
+
+    deprecated_versions.include?(version)
+  end
+
+  sig { override.returns(T::Boolean) }
+  def unsupported?
+    # Check if the feature flag for Bundler v1 unsupported error is enabled.
+    return false unless Dependabot::Experiments.enabled?(:bundler_v1_unsupported_error)
```

Comment:
The feature flag going to be activated when we are not supporting the version anymore.